### PR TITLE
Add properties for destinations in XSJS API

### PR DESCRIPTION
**Pull request by d-velev**
This PR makes it so that a destination wrapper object is returned in `getDestination`, in which we store host, port, pathPrefix and any other properties that the `cloudplatform.connectivity.Destination` object has (along with custom properties).

### CODE REVIEW

**Review by MarinHadzhiev — CHANGES_REQUESTED** [review]
Please replace sonar.sources property in the module pom with

<sonar.sources>src/main/resources/META-INF/dirigible/xsk,src/main/java/</sonar.sources>

so that also the java sources are analysed by the Sonar

**Review by MarinHadzhiev — COMMENTED** [review]
missing test for DestMailConfigProvider (that is why sonar check is failing)

File: modules/api/api-xsjs/src/test/java/com/sap/xsk/api/test/CloudPlatformDestinationFacadeTest.java [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+@RunWith(Parameterized.class)
+public class CloudPlatformDestinationFacadeTest {
+
+  HttpClient httpClient;
+  String destinationName = "test-destination";
+  String DESTINATION_URI = "http://test-destination.com:8080/destination";
+  MockedStatic<DestinationAccessor> destinationAccessor;
+  MockedStatic<HttpClientAccessor> httpClientAccessor;
+  MockedStatic<CloudPlatformAccessor> cloudPlatformAccessor;
+  boolean isKymaFacadeSet = false;
```
Comment (ThuF) [inline modules/api/api-xsjs/src/test/java/com/sap/xsk/api/test/CloudPlatformDestinationFacadeTest.java:56]: Please use the `private` modifier

File: modules/engines/engine-mail-destination/pom.xml [hunk 2]
```diff
@@ -32,6 +32,30 @@
...
+      <artifactId>mockito-inline</artifactId>
+      <version>${mockito.version}</version>
+      <scope>test</scope>
+    </dependency>
+    <dependency>
+      <groupId>junit</groupId>
+      <artifactId>junit</artifactId>
+      <version>RELEASE</version>
+      <scope>test</scope>
+    </dependency>
```
Comment (ThuF) [inline modules/engines/engine-mail-destination/pom.xml:58]: Not needed?

File: modules/engines/engine-mail-destination/src/test/java/com/sap/xsk/mail/test/DestMailConfigProviderTest.java [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Mockito.when;
+
+public class DestMailConfigProviderTest {
+
+  private static final String MAIL_SERVER_DESTINATION_NAME = "MAIL_SERVER_DESTINATION_NAME";
+  private static final String DESTINATION_NAME = "test-destination";
+
+  MockedStatic<CloudPlatformDestinationFacade> cloudPlatformDestinationFacadeMock;
+  MockedStatic<Configuration>configurationMock;
```
Comment (ThuF) [inline modules/engines/engine-mail-destination/src/test/java/com/sap/xsk/mail/test/DestMailConfigProviderTest.java:33]: `private`?

File: modules/api/api-xsjs/src/main/java/com/sap/xsk/api/destination/Destination.java [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+import java.util.Properties;
+
+public class Destination {
+
+  private String host;
+  private int port;
+  private String pathPrefix;
+  private Properties properties = new Properties();
+
+  Destination(String host, int port, String pathPrefix, Properties properties) {
```
Comment (ThuF) [inline modules/api/api-xsjs/src/main/java/com/sap/xsk/api/destination/Destination.java:24]: `public Destination(...)`?

File: modules/api/api-xsjs/src/main/resources/META-INF/dirigible/xsk/http/http.js [hunk 5]
```diff
@@ -69,9 +69,17 @@ var SET_COOKIE_HEADER = "Set-Cookie";
...
+  var readDestination = com.sap.xsk.api.destination.CloudPlatformDestinationFacade.getDestination(destinationName);
+  var destination = new exports.Destination();
 
-  return destination;
+  destination.host = readDestination.getHost();
+  destination.port = readDestination.getPort();
+  destination.pathPrefix = readDestination.getPathPrefix();
+  destination.name = destinationName;
+
+  var properties = JSON.parse(readDestination.getPropertiesAsJSON());
```
Comment (ThuF) [inline modules/api/api-xsjs/src/main/resources/META-INF/dirigible/xsk/http/http.js:80]: `let` instead of `var`?

**Review by ThuF — CHANGES_REQUESTED** [review]
Just few minor comments to be addressed before merging t he PR.

File: modules/engines/engine-mail-destination/src/test/java/com/sap/xsk/mail/test/DestMailConfigProviderTest.java [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+
+  @After
+  public void close() {
+    configurationMock.close();
+    cloudPlatformDestinationFacadeMock.close();
+  }
+
+  public void mock() {
+    Properties props = new Properties();
+    props.setProperty("mail.user", "dirigible@eclipse.com");
```
Comment (MarinHadzhiev) [inline modules/engines/engine-mail-destination/src/test/java/com/sap/xsk/mail/test/DestMailConfigProviderTest.java:48]: change the user to something not related to OSS component

**Review by MarinHadzhiev — APPROVED** [review]

**Review by ThuF — APPROVED** [review]
Seems good to me

**Pull request merged**